### DISCUSSION TITLE
Adding known database limitations to Directus website

### DISCUSSION THREAD

**Discussion by paulboudewijn** [directus/directus#9679] [Feature Requests]
The more I work with Directus, the more enthusiastic I become. BUT, on the other hand, the more I run into database limitations.
The Directus website states "pure SQL queries" and "no restrictions", but there are certainly restrictions and AFAIK those are not documented anywhere yet.
I think in order to properly manage the expectations of (new) users, especially those who want to use an existing database, some kind of database specific content on the directus website would be helpful.

So far I have encountered the following limitations:

**MSSQL:**
- Only `dbo` scheme is supported; no workaround available.
- Triggers are not supported; content in tables with triggers on them cannot be edited via Directus, because of the use of the OUTPUT clause (by the Directus API). Workaround is the use of API hooks, but documentation is limited.

**MySQL**
- Tinyint(0) and Tinyint(1) are always considered booleans;

Any thoughts on this?

**Comment by benhaynes** [comment]
This probably won't go in the website, but we could add a section to the Docs about the nuance of database abstraction across vendors with different feature support, and edge cases that are not yet covered. I imagine these would go in as warnings within the database mirroring section. Since that section is so important, and is growing in size, we might want to make it a dedicated page within Getting Started.

https://docs.directus.io/getting-started/introduction/#database-mirroring

```
::: warning MS-SQL Limitations

* Only `dbo` scheme is supported; no workaround available.
* Triggers are not supported; content in tables with triggers on them cannot be edited via Directus, because of the use of the OUTPUT clause (by the Directus API). Workaround is the use of API hooks, but documentation is limited.

:::
```

Do you want to give a PR a try for these warnings?

**Reply by paulboudewijn** [reply]
Done! PR #9702

**Comment by rijkvanzanten** [comment]
> The Directus website states "pure SQL queries" and "no restrictions", but there are certainly restrictions and AFAIK those are not documented anywhere yet.

Mind detailing where / what exactly? Context definitely matters here 

Also wondering if we should list bugs/open feature requests of Directus itself as "limitations" 🤔  Like you're free to use triggers on the database itself, but there's no way to set that up in Directus itself yet. Is that a limitation, or simply a feature that doesn't exist yet? Is there a difference between the two? Does it matter for the 80/20? Where do we draw the limit with that? For example, should we consider things like the fact that you can generate thumbnails for images but not for videos yet a limitation or a feature that doesn't exist yet?

Generally speaking, I'd prefer listing the things that we _do_ support, with the knowledge that everything that isn't explicitly documented isn't supported.

**Reply by paulboudewijn** [reply]
> Like you're free to use triggers on the database itself, but there's no way to set that up in Directus itself yet.

No, the problem is content in tables with triggers on them can't be edited via Directus. Not being able to setup triggers via Directus wouldn't be a problem, but this truely is.

**Reply by paulboudewijn** [reply]
> Generally speaking, I'd prefer listing the things that we _do_ support, with the knowledge that everything that isn't explicitly documented isn't supported.

Fully agree, but then the website shouldn't say "no restrictions".

**Reply by rijkvanzanten** [reply]
> No, the problem is content in tables with triggers on them can't be edited via Directus.

That sounds like a bug to me, not a known limitation that's there on purpose 👍🏻 

> Fully agree, but then the website shouldn't say "no restrictions".

Mind detailing where exactly? Context definitely matters here.

If you're talking about the following on https://directus.io/open-source/:

> Our premium software is available at no cost for commercial and personal use. This self-hosted version is full-featured, with **no artificial limitations**.

(emphasis mine) then I think it's appropriate. We don't put any artificial limitations on using the open-source version as compared to the cloud offering. That being said, I might be misunderstanding what bit you're talking about exactly 👍🏻

**Reply by benhaynes** [reply]
> Fully agree, but then the website shouldn't say "no restrictions".

This is certainly a gray area, as there are subtle differences between restrictions, intrinsic limitations, bugs, and features that haven't been built yet. We have not _built a restriction_ regarding this, it's just an incomplete part of the platform (and/or an artifact of the database vendor abstraction).

In any event, it's worth clarifying in the Docs.

**Reply by paulboudewijn** [reply]
> Mind detailing where exactly? Context definitely matters here.

![image](https://user-images.githubusercontent.com/17253180/141159029-7f77f9f8-5e60-4150-a58f-0f125317b808.png)

![image](https://user-images.githubusercontent.com/17253180/141187158-5f18c101-2c94-46b0-93d9-12a45491e853.png)

**Reply by paulboudewijn** [reply]
> That sounds like a bug to me, not a known limitation that's there on purpose 👍🏻

I'll create an issue (#9704) on this one, nevertheless I think it's worth mentioning to new users that this 'limition' currently exists.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
